tests/clipboard.spec.ts

```diff
@@ -139,6 +141,56 @@ test(
   }
 );
 
+test(
+  scoped`clipboard paste end with image, the cursor should be controlled by up/down keys`,
+  async ({ page }) => {
+    test.info().annotations.push({
+      type: 'issue',
+      description: 'https://github.com/toeverything/blocksuite/issues/3639',
+    });
+    await enterPlaygroundRoom(page);
+    await initEmptyParagraphState(page);
+    await focusRichText(page);
+
+    // set up clipboard data using html
+    const clipData = {
+      'text/html': `<p>符合 Markdown 格式的 URL 放到笔记中，此时需要的格式如下：</p>
```

Comment:
Would be better to use English as test case content, unless it's related to IME or emoji.